packages/sidecar/src/messageBuffer.ts

```diff
@@ -30,16 +30,36 @@ export class MessageBuffer<T> {
       if (atItem[0] > minTime) break;
       this.head += 1;
     }
+
+    for (const readerId of this.readers.keys()) {
+      const readerInfo = this.readers.get(readerId);
+      if (!readerInfo) continue;
```

Comment:
This should never happen? Also, we should be using [`Map.prototype.entries()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/entries):

```ts
for (const [readerId, readerInfo] of this.readers.entries()) {
```